NotesApi/V1/Gateways/INotesApiGateway.cs

```diff
@@ -1,12 +1,12 @@
+using System;
 using System.Collections.Generic;
+using System.Threading.Tasks;
 using NotesApi.V1.Domain;
 
 namespace NotesApi.V1.Gateways
 {
     public interface INotesApiGateway
     {
-        Entity GetEntityById(int id);
-
-        List<Entity> GetAll();
+        Task<List<Note>> GetByTargetIdAsync(Guid targetId);
```

Comment:
As rule of thumb, it would be good if we use IEnumerable instead of strongly type List or Array